clientgenv2/source_generator.go

```diff
@@ -130,6 +211,25 @@ func (r *SourceGenerator) NewResponseField(selection ast.Selection, typeName str
 			// if a child field is fragment, this field type became fragment.
 			baseType = fieldsResponseFields[0].Type
 		case fieldsResponseFields.IsStructType():
+			// 子フィールドにFragmentがある場合は、現在のフィールドとマージする
+			// if there is a fragment in child fields, merge it with the current field
+			generator := NewStructGenerator(fieldsResponseFields)
+
+			// remove pre-merged struct
+			for _, preMergedTypeName := range generator.GetPreMergedStructSources() {
+				for i, source := range r.StructSources {
+					// when name is same, remove it
+					if source.Name == preMergedTypeName.Name {
+						r.StructSources = append(r.StructSources[:i], r.StructSources[i+1:]...)
+						break
+					}
+				}
+			}
```

Comment:
Pretty good Fix! In my opinion, the best implementation would be something like this:

```suggestion
　　　　　　　r.StructSources = generator.MargedStructSources(r.StructSources)
```

Reply:
This is pretty nice advice. I will try!